product_cost_usd/models/product_template.py

```diff
@@ -0,0 +1,41 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    standard_price_usd = fields.Float(
+        "Cost in USD", digits="Product Price", help="Price cost of the product in USD currency"
+    )
+
+    @api.constrains("standard_price_usd", "seller_ids")
+    def check_cost_and_price(self):
+        """Validate 'Cost in USD' usability.
+
+        Usability conditions:
+        - Before set a 'Cost in USD' in a product at least one supplier should
+          have price in USD.
+        - The Cost in USD cannot be less than supplier price.
+        """
+        usd_currency = self.env.ref("base.USD")
+        for product in self:
+            prec = product.env["decimal.precision"].precision_get("Product Price")
+            usd_seller = product.seller_ids.filtered(lambda x: x.currency_id == usd_currency)
+            list_price = usd_seller.price
+            standard_price_usd = product.standard_price_usd
+            if not usd_seller and float_compare(standard_price_usd, 0, precision_digits=prec) > 0:
+                raise ValidationError(
+                    _("You must have at least one supplier with price in USD before assigning a Cost in USD")
+                )
+            if float_compare(list_price, standard_price_usd, precision_digits=prec) > 0:
+                raise ValidationError(
+                    _(
+                        "You cannot create or modify a product if the cost in USD"
+                        " is less than the supplier list price.\n\n"
+                        "- Supplier list price = %s\n"
+                        "- Cost in USD = %s",
+                        list_price, standard_price_usd
```

Comment:
This semmseems to not be the black format:

```suggestion
                        list_price,
                        standard_price_usd,
```